### ISSUE TITLE
TreemapView with D3.js

### ISSUE THREAD

**Issue by JohannesLichtenberger** [sirixdb/sirix-web-frontend#14]
In order to get an overview about the stored XML and JSON resources we might want to use space filling approaches for tree-structured data as for instancdd a TreeMap.

**Comment by MartinTuroci** [comment]
Where can I find data to be displayed? Can you provide some endpoint?

**Comment by JohannesLichtenberger** [comment]
Hm, can you use dummy data for now? Simply JSON and XML :-)

**Comment by JohannesLichtenberger** [comment]
Or you could try to get the HTTP-server up and running using the docker-compose file in the bundle `sirix-rest-api`. Some documentation: https://sirix.io/rest-api.html

However, the binary content simply gets serialized as JSON or XML again, depending on the database type (what content is stored).

**Comment by MartinTuroci** [comment]
ok, currently I am setting up keycloak via docker (according to README). The link you provided for documentation talks nothing about docker though. It would be great if you could build a single image, push it to the repository so that all the frontend (and backend of course) guys need to run just 1 command and they are up and running.

**Comment by JohannesLichtenberger** [comment]
Yes, I think the only thing missing in the docker-compose file is the realm data, which I have exported from my running Keycloak instance to provide a simple test configuration. Will do this next, but maybe tomorrow.

`docker-compose up -d` hopefully. Need to check this on my laptop at work. Here I have only Windows, but maybe Docker for Windows is now a bit better than a few years ago...

You can import it via the Keycloak web frontend however:

https://github.com/sirixdb/sirix/tree/master/bundles/sirix-rest-api/src/test/resources

**Comment by JohannesLichtenberger** [comment]
The databases are per default stored in `$USER_HOME/sirix-data`

**Comment by JohannesLichtenberger** [comment]
Oh and to keep you informed, simonholdorf from dev.to would like to switch to a Nuxt.js setup :-)

**Comment by MartinTuroci** [comment]
Okay, so I will have to probably wait with all the tasks till he does the migration. Why did you decide to make a switch from SPA to serverside rendering? (just curious :) )

**Comment by JohannesLichtenberger** [comment]
I think preloading on the server adds a lot of value. Otherwise it's still a PWA, right? :-)

**Comment by simhol** [comment]
With nuxt we'll combine the power of both worlds. It's built as a universal app where the initial HTML and other static content is rendered by the server but as soon as something happens in the client it will be handled as a SPA by Vue. This has great effects on performance, SEO

**Comment by simhol** [comment]
I've already set it up the nuxt project with the state of this repo

**Comment by simhol** [comment]
https://github.com/simhol/sirix-nuxt-frontend 

If you like it you could integrate it into your repo. I hope i migrated everything - if not, just holla at me :)

**Comment by JohannesLichtenberger** [comment]
Very cool, thanks a lot. Regarding the structure: Can we split the view according to the latest PR?

https://github.com/sirixdb/sirix-web-frontend/tree/master/src

With dependency injection and have a look at the `models`, `views`... there's also a navigation bar component added :-)

**Comment by JohannesLichtenberger** [comment]
Then you could make a PR to integrate it, would be great :-)

**Comment by JohannesLichtenberger** [comment]
@MartinTuroci, we migrated :-)

**Comment by MartinTuroci** [comment]
Good job guys. I am currently quite busy, hopefully will be able to have a look by the end of the week.

**Comment by JohannesLichtenberger** [comment]
Sounds great, have a nice week :-)
